src/controller/java/src/chip/devicecontroller/model/AttributeWriteRequest.java

```diff
@@ -20,12 +20,19 @@
 import java.util.Locale;
 import java.util.Objects;
 import java.util.Optional;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.annotation.Nullable;
+import org.json.JSONException;
+import org.json.JSONObject;
 
 /** An attribute write request that should be used for interaction model write interaction. */
 public final class AttributeWriteRequest {
+  private static final Logger logger = Logger.getLogger(AttributeWriteRequest.class.getName());
   private final ChipPathId endpointId, clusterId, attributeId;
   private final Optional<Integer> dataVersion;
-  private final byte[] tlv;
+  @Nullable private final byte[] tlv;
+  @Nullable private final JSONObject json;
```

Comment:
these two should be mutual exclusive, what if the user provide both

Reply:
please see the public function, which has mutual exclusive, done, thanks

Reply:
tlv and json are not mutual exclusive for invoke, since they could be both absent for invoke response. thanks